### DISCUSSION TITLE
Custom Visualization for Superset with hooks

### DISCUSSION THREAD

**Discussion by sumosuzy** [apache/superset#18053] [Q&A / Help]
I am working on custom visualizations for Superset

I have followed the following instructions
https://superset.apache.org/docs/installation/building-custom-viz-plugins

In the example on the page choose the class component instead of function component (with hooks)

I follow through and I get the chart but I get the following error - ooks can only be called inside of the body of a function component

It looks like follows

![image](https://user-images.githubusercontent.com/58918955/149572670-03c43398-3eeb-4510-8f3a-0a4c1b221b87.png)

Has anyone managed to get this to work?   I think it is because the function with hooks is being called from a  class

**Comment by srinify** [comment]
Hey there, I can't speak to the specific issue you're running into ... but @villebro recently updated the custom viz plugin implementation here.

- He wrote about it here --  https://preset.io/blog/building-custom-viz-plugins-in-superset-v2/
- He's also hosting a Meetup / Event on this! -- https://preset.io/events

**Reply by sumosuzy** [reply]
Thank you very much. I am going to try to get time off work so I can join the meet up,  I have gotten as far as he has in his blog and have one bar plug in that I did and the beginnings of a map one.  However if I try to generate the code as in the blog using the generator superset generator as pictured in the blog the chart plug in with hooks option will not work.   The delivered code with not changes for the hello world generates a hooks can only be called inside of the body of a function component when you open the chart in superset with a dataset
